### DISCUSSION TITLE
Best practices to perform clinical trial simulations

### DISCUSSION THREAD

**Discussion by mark0935git** [metrumresearchgroup/mrgsolve#984] [Q&A] [answered]
Dear Mrgsolve community,

I would like to know what is the best practices to perform clinical trial simulations?

We can certainly write a loop to complete the task of "performing 100 simulated clinical trials to determine the summary exposures". But I am wondering whether there are smarter ways for the task?

If so, could you please give me or point me to some instructions?

Many thanks in advance!

**Comment by kylebaron** [comment]
Hi @mark0935git - 

The topic of clinical trial simulation can be pretty broad, including thinking about generating the clinical trial population, propagating uncertainty in model parameters, dropout, analysis etc. 

Is there more information you could share that could prompt some more specific types of feedback on the question?

Regarding this question:
>We can certainly write a loop to complete the task

That is essentially now we usually do clinical trial simulation.  Not necessarily a loop, but using  `lapply()` or `map()` and possibly in parallel. If the setup is complicated, I might write some R code to generate or resample the population etc. But it really depends on what your needs are.

Let's keep the conversation going. 

Kyle

**Comment by smouksassi** [comment, marked as answer]
I try to do this here 
https://github.com/smouksassi/coveffectsplot/blob/main/vignettes/PK_Example_full.Rmd#L411
technically not a full blown CTS but it incorporates full distribution of covariates and uncertainty
(we can also add uncertainty on BSV if need be...)
